From 9fd478fb4b61077bb435702bcb65e79de681e1c6 Mon Sep 17 00:00:00 2001
From: Axel Lin <axel.lin@gmail.com>
Date: Sat, 26 Nov 2011 12:55:43 +0800
Subject: mmc: convert drivers/mmc/host/* to use module_platform_driver()
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

This patch converts the drivers in drivers/mmc/host/* to use the
module_platform_driver() macro which makes the code smaller and a bit
simpler.

Signed-off-by: Axel Lin <axel.lin@gmail.com>
Acked-by: "Michał Mirosław" <mirq-linux@rere.qmqm.pl>
Acked-by: David Brown <davidb@codeaurora.org>
Acked-by: Viresh Kumar <viresh.kumar@st.com>
Acked-by: Guennadi Liakhovetski <g.liakhovetski@gmx.de>
Acked-by: Sascha Hauer <s.hauer@pengutronix.de>
Acked-by: Wolfram Sang <w.sang@pengutronix.de>
Acked-by: Anton Vorontsov <cbouatmailru@gmail.com>
Signed-off-by: Chris Ball <cjb@laptop.org>
(cherry-picked from commit d1f81a64a4250bdd776978be06ae2b8e13ec7471)

Conflicts:

	drivers/mmc/host/bfin_sdh.c
	drivers/mmc/host/cb710-mmc.c
	drivers/mmc/host/jz4740_mmc.c
	drivers/mmc/host/msm_sdcc.c
	drivers/mmc/host/mxcmmc.c
	drivers/mmc/host/mxs-mmc.c
	drivers/mmc/host/pxamci.c
	drivers/mmc/host/s3cmci.c
	drivers/mmc/host/sdhci-cns3xxx.c
	drivers/mmc/host/sdhci-dove.c
	drivers/mmc/host/sdhci-esdhc-imx.c
	drivers/mmc/host/sdhci-of-esdhc.c
	drivers/mmc/host/sdhci-of-hlwd.c
	drivers/mmc/host/sdhci-pxav2.c
	drivers/mmc/host/sdhci-pxav3.c
	drivers/mmc/host/sdhci-s3c.c
	drivers/mmc/host/sdhci-spear.c
	drivers/mmc/host/sdhci-tegra.c
	drivers/mmc/host/sh_mmcif.c

Signed-off-by: Simon Horman <horms@verge.net.au>
---
 drivers/mmc/host/sh_mobile_sdhi.c | 13 +------------
 drivers/mmc/host/tmio_mmc.c       | 14 +-------------
 2 files changed, 2 insertions(+), 25 deletions(-)

diff --git a/drivers/mmc/host/sh_mobile_sdhi.c b/drivers/mmc/host/sh_mobile_sdhi.c
index 41ae646..58da3c4 100644
--- a/drivers/mmc/host/sh_mobile_sdhi.c
+++ b/drivers/mmc/host/sh_mobile_sdhi.c
@@ -282,18 +282,7 @@ static struct platform_driver sh_mobile_sdhi_driver = {
 	.remove		= __devexit_p(sh_mobile_sdhi_remove),
 };
 
-static int __init sh_mobile_sdhi_init(void)
-{
-	return platform_driver_register(&sh_mobile_sdhi_driver);
-}
-
-static void __exit sh_mobile_sdhi_exit(void)
-{
-	platform_driver_unregister(&sh_mobile_sdhi_driver);
-}
-
-module_init(sh_mobile_sdhi_init);
-module_exit(sh_mobile_sdhi_exit);
+module_platform_driver(sh_mobile_sdhi_driver);
 
 MODULE_DESCRIPTION("SuperH Mobile SDHI driver");
 MODULE_AUTHOR("Magnus Damm");
diff --git a/drivers/mmc/host/tmio_mmc.c b/drivers/mmc/host/tmio_mmc.c
index a4ea102..113ce6c 100644
--- a/drivers/mmc/host/tmio_mmc.c
+++ b/drivers/mmc/host/tmio_mmc.c
@@ -138,19 +138,7 @@ static struct platform_driver tmio_mmc_driver = {
 	.resume = tmio_mmc_resume,
 };
 
-
-static int __init tmio_mmc_init(void)
-{
-	return platform_driver_register(&tmio_mmc_driver);
-}
-
-static void __exit tmio_mmc_exit(void)
-{
-	platform_driver_unregister(&tmio_mmc_driver);
-}
-
-module_init(tmio_mmc_init);
-module_exit(tmio_mmc_exit);
+module_platform_driver(tmio_mmc_driver);
 
 MODULE_DESCRIPTION("Toshiba TMIO SD/MMC driver");
 MODULE_AUTHOR("Ian Molton <spyro@f2s.com>");
-- 
1.7.10.2.565.gbd578b5

